### ISSUE TITLE
 [CI] MixedClusterClientYamlTestSuiteIT 'Create a snapshot and then restore it' failures

### ISSUE THREAD

**Issue by davidkyle** [elastic/elasticsearch#64152]
**Build scan**:
https://gradle-enterprise.elastic.co/s/m62agbvicmhiy
https://gradle-enterprise.elastic.co/s/74gmzvgqiplla
https://gradle-enterprise.elastic.co/s/psp5un562yyzo

**Repro line**:
./gradlew ':qa:mixed-cluster:v7.0.0#mixedClusterTest' --tests "org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT" \
  -Dtests.method="test {p0=snapshot.restore/10_basic/Create a snapshot and then restore it}" \
  -Dtests.seed=C48535D3AF9BDB85 \
  -Dtests.security.manager=true \
  -Dtests.locale=fr \
  -Dtests.timezone=America/Denver \
  -Druntime.java=8

**Reproduces locally?**:
No

**Applicable branches**:
7.x, 7.10

**Failure history**:
Quite a few since 2020/10/02

https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-30d,mode:quick,to:now))&_a=(columns:!(build.branch,build.build-scans),index:b646ed00-7efc-11e8-bf69-63c8ef516157,interval:auto,query:(language:lucene,query:'%22Create%20a%20snapshot%20and%20then%20restore%20it%22'),sort:!(process.time-start,desc))

**Failure excerpt**:
```
java.lang.AssertionError: Failure at [snapshot.restore/10_basic:54]: field [test_index.shards.0.type] is null
	at __randomizedtesting.SeedInfo.seed([C48535D3AF9BDB85:4CD10A090167B67D]:0)
	at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:414)
	at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.test(ESClientYamlSuiteTestCase.java:391)
```

**Comment by davidkyle** [comment]
Muted 7.x: 786c5156341479dc04dc0e3557014de6ab653c53 and 7.10: 4545779415e837610c0db61c9668b197c92bafe7

**Comment by original-brownbear** [comment]
This looks like it was caused by adding the new `index.uuid` setting and messing up in terms of BwC in https://github.com/elastic/elasticsearch/pull/56930:

```
» WARN ][o.e.c.r.a.AllocationService] [v7.0.0-0] failing shard [IllegalArgumentException[unknown setting [index.history.uuid] please check that any required plugins are installed, or check the breaking changes documentation for removed settings]], markAsStale [true]]
»  java.lang.IllegalArgumentException: unknown setting [index.history.uuid] please check that any required plugins are installed, or check the breaking changes documentation for removed settings
»  	at org.elasticsearch.common.settings.AbstractScopedSettings.validate(AbstractScopedSettings.java:531) ~[elasticsearch-7.11.0-SNAPSHOT.jar:7.11.0-SNAPSHOT]
»  	at org.elasticsearch.common.settings.AbstractScopedSettings.validate(AbstractScopedSettings.java:476) ~[elasticsearch-7.11.0-SNAPSHOT.jar:7.11.0-SNAPSHOT]
»  	at org.elasticsearch.common.settings.AbstractScopedSettings.validate(AbstractScopedSettings.java:447) ~[elasticsearch-7.11.0-SNAPSHOT.jar:7.11.0-SNAPSHOT]
»  	at org.elasticsearch.indices.IndicesService.createIndexService(IndicesService.java:529) ~[elasticsearch-7.11.0-SNAPSHOT.jar:7.11.0-SNAPSHOT]
»  	at org.elasticsearch.indices.IndicesService.createIndex(IndicesService.java:493) ~[elasticsearch-7.11.0-SNAPSHOT.jar:7.11.0-SNAPSHOT]
»  	at org.elasticsearch.indices.IndicesService.createIndex(IndicesService.java:161) ~[elasticsearch-7.11.0-SNAPSHOT.jar:7.11.0-SNAPSHOT]
»  	at org.elasticsearch.indices.cluster.IndicesClusterStateService.createIndices(IndicesClusterStateService.java:498) ~[elasticsearch-7.11.0-SNAPSHOT.jar:7.11.0-SNAPSHOT]
»  	at org.elasticsearch.indices.cluster.IndicesClusterStateService.applyClusterState(IndicesClusterStateService.java:268) ~[elasticsearch-7.11.0-SNAPSHOT.jar:7.11.0-SNAPSHOT]
»  	at org.elasticsearch.cluster.service.ClusterApplierService.lambda$callClusterStateAppliers$5(ClusterApplierService.java:478) ~[elasticsearch-7.11.0-SNAPSHOT.jar:7.11.0-SNAPSHOT]
»  	at java.lang.Iterable.forEach(Iterable.java:75) ~[?:1.8.0_241]
»  	at org.elasticsearch.cluster.service.ClusterApplierService.callClusterStateAppliers(ClusterApplierService.java:476) ~[elasticsearch-7.11.0-SNAPSHOT.jar:7.11.0-SNAPSHOT]
»  	at org.elasticsearch.cluster.service.ClusterApplierService.applyChanges(ClusterApplierService.java:459) ~[elasticsearch-7.11.0-SNAPSHOT.jar:7.11.0-SNAPSHOT]
»  	at org.elasticsearch.cluster.service.ClusterApplierService.runTask(ClusterApplierService.java:413) ~[elasticsearch-7.11.0-SNAPSHOT.jar:7.11.0-SNAPSHOT]
»  	at org.elasticsearch.cluster.service.ClusterApplierService$UpdateTask.run(ClusterApplierService.java:164) ~[elasticsearch-7.11.0-SNAPSHOT.jar:7.11.0-SNAPSHOT]
»  	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:681) [elasticsearch-7.11.0-SNAPSHOT.jar:7.11.0-SNAPSHOT]
»  	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:252) [elasticsearch-7.11.0-SNAPSHOT.jar:7.11.0-SNAPSHOT]
»  	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:215) [elasticsearch-7.11.0-SNAPSHOT.jar:7.11.0-SNAPSHOT]
»  	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:1.8.0_241]
»  	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:1.8.0_241]
»  	at java.lang.Thread.run(Thread.java:835) [?:1.8.0_241]
»   ↑ repeated 5 times ↑
```

looking into a fix

**Comment by original-brownbear** [comment]
Fixed by #64164
